inc/media.php

```diff
@@ -2055,40 +2055,37 @@ function media_nstree_li($item){
  */
 function media_resize_image($file, $ext, $w, $h=0){
     global $conf;
-
-    $info = @getimagesize($file); //get original size
-    if($info == false) return $file; // that's no image - it's a spaceship!
-
-    if(!$h) $h = round(($w * $info[1]) / $info[0]);
-    if(!$w) $w = round(($h * $info[0]) / $info[1]);
-
+    if(!$h) $h = $w;
     // we wont scale up to infinity
     if($w > 2000 || $h > 2000) return $file;
 
-    // resize necessary? - (w,h) = native dimensions
-    if(($w == $info[0]) && ($h == $info[1])) return $file;
-
     //cache
     $local = getCacheName($file,'.media.'.$w.'x'.$h.'.'.$ext);
     $mtime = @filemtime($local); // 0 if not exists
 
-    if($mtime > filemtime($file) ||
-        media_resize_imageIM($ext, $file, $info[0], $info[1], $local, $w, $h) ||
-        media_resize_imageGD($ext, $file, $info[0], $info[1], $local, $w, $h)
-    ) {
-        if($conf['fperm']) @chmod($local, $conf['fperm']);
-        return $local;
+    $options = [
+        'quality' => $conf['jpg_quality'],
+        'imconvert' => $conf['im_convert'],
+    ];
+
+    if( $mtime > @filemtime($file) ) {
```

Comment:
This is broken. It should be `$mtime <= (int) @filemtime($file)`. See my patch here: https://github.com/splitbrain/dokuwiki/pull/3259

Reply:
hah. you're right of course. Thank you